CommandLineClient/src/test/java/com/dslplatform/compiler/client/MonoRuntimeErrorDetectionTest.java

```diff
@@ -0,0 +1,55 @@
+package com.dslplatform.compiler.client;
+
+import com.dslplatform.compiler.client.parameters.DslCompiler;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+
+public class MonoRuntimeErrorDetectionTest {
+
+    @Test
+    public void testSegmentationFault() {
+        Assert.assertTrue(executeMonoNativeFailureTest((fileContent("/segmentationFault.txt"))));
+    }
+
+    @Test
+    public void testInvalidILCode() {
+        Assert.assertTrue(executeMonoNativeFailureTest(fileContent("/invalidILCode.txt")));
+    }
+
+    @Test
+    public void testValidOutput() {
+        Assert.assertFalse(executeMonoNativeFailureTest((fileContent("/validOutput.txt"))));
+    }
+
+
+    private static Boolean executeMonoNativeFailureTest(String output) {
+        try {
+            Method monoNativeFailure = DslCompiler.class.getDeclaredMethod("monoNativeFailure", String.class);
```

Comment:
Yeah, that's another way. I dislike making things package-private just to be able to test them.